[Bug]: ChainService stops execution even when continueOnError = true

### Bug Description

When executing a chain, if a step fails but has continueOnError = true, the ChainService unexpectedly stops executing remaining steps and marks them as "skipped".
Expected behavior is to continue executing subsequent steps.

### Steps to Reproduce

1.Create a chain with two steps.

2.Step 1: Simulate failure (mock httpClient to throw an error).

3.Step 1: Set continueOnError = true.

4.Step 2: Mock successful execution.

5.Run chainService.executeChain().

6.Observe the execution result.

### Expected Behavior

Step 1: "failed"

Step 2: "success"

### Actual Behavior

Step 1: "failed"

Step 2: "skipped"

### Environment

Vitest/ui v4.0.5
Vitest/coverage-v8 v4.0.4

Node.js v22.2.0

OS: Windows 11

File: src/lib/chain-service.ts

Test file: chain-service.test.ts

### Screenshots

_No response_

### Additional Context

AssertionError: expected 'skipped' to be 'success'

--- a/e2e/object-models/api-model.ts
+++ b/e2e/object-models/api-model.ts
@@ -20,10 +20,10 @@ export class APImodel extends  ReqHelpers {
                                                         .getByRole('textbox') 
         
 
-    async get(url:string):Promise<string> 
+    async get(url:string):Promise<void>
     {
         await this.fillUrl.fill(url)
-        
+
         await this.reqType.selectOption('GET')
     }
 
@@ -51,7 +51,7 @@ export class APImodel extends  ReqHelpers {
         await this.page.waitForLoadState('networkidle')
         await this.responseBody.locator('.view-line').last().textContent()
         let result = await this.responseBody.textContent()
-        result = result.replace(/\u00A0/g, ' ')
+        result = result?.replace(/\u00A0/g, ' ') || ''
         return result
     }
 

--- a/e2e/object-models/request-assiters.model.ts
+++ b/e2e/object-models/request-assiters.model.ts
@@ -78,7 +78,7 @@ export class ReqHelpers extends BasePage {
     
     async uncheckHeader(hName:string):Promise<Locator> {
         await this.checkIfInheaderSection()
-        let cb: Locator
+        let cb: Locator | undefined
         const placeHolder :string = (await this.reqHeaderName.getAttribute('placeholder'))!
         const allph =  await this.reqBuilderMain.getByPlaceholder(placeHolder).all()
         for(let i=0; i < allph.length; i++) {
@@ -86,7 +86,9 @@ export class ReqHelpers extends BasePage {
                 cb = await this.reqBuilderMain.getByRole('checkbox').nth(i)
                 await cb.uncheck()
             }
-        } return cb
+        }
+        if (!cb) throw new Error(`Header ${hName} not found`)
+        return cb
     }
     
     async deleteHeader(hName: string) {

--- a/package.json
+++ b/package.json
@@ -31,6 +31,7 @@
     "@playwright/test": "^1.56.1",
     "@testing-library/jest-dom": "^6.9.1",
     "@testing-library/react": "^16.3.0",
+    "@types/dotenv": "^6.1.1",
     "@types/node": "^22",
     "@types/react": "^19",
     "@types/react-dom": "^19",

--- a/src/components/ComparisonSelector.tsx
+++ b/src/components/ComparisonSelector.tsx
@@ -115,9 +115,9 @@ export default function ComparisonSelector({ isOpen, onClose, onSelect }: Compar
                         {item.method}
                       </span>
                       <span className={`px-2 py-0.5 text-xs rounded ${
-                        item.response?.status >= 200 && item.response?.status < 300
+                        item.response?.status !== undefined && item.response.status >= 200 && item.response.status < 300
                           ? 'bg-green-100 text-green-700 dark:bg-green-900/20 dark:text-green-400'
-                          : item.response?.status >= 400
+                          : item.response?.status !== undefined && item.response.status >= 400
                           ? 'bg-red-100 text-red-700 dark:bg-red-900/20 dark:text-red-400'
                           : 'bg-gray-100 text-gray-700 dark:bg-gray-800 dark:text-gray-400'
                       }`}>

--- a/src/components/WebSocketPanel.tsx
+++ b/src/components/WebSocketPanel.tsx
@@ -22,7 +22,7 @@ export default function WebSocketPanel() {
       }
     });
 
-    return () => unsubscribe();
+    return () => { unsubscribe(); };
   }, []);
 
   useEffect(() => {

--- a/src/lib/chain-service.ts
+++ b/src/lib/chain-service.ts
@@ -230,14 +230,13 @@ class ChainService {
 
           // Check if we should continue on error
           if (!step.continueOnError) {
-            throw new Error(`Step ${i + 1} failed: ${error.message}`);
+            // Mark remaining steps as skipped
+            for (let j = i + 1; j < execution.steps.length; j++) {
+              execution.steps[j].status = 'skipped';
+            }
+            break;
           }
-
-          // Mark remaining steps as skipped
-          for (let j = i + 1; j < execution.steps.length; j++) {
-            execution.steps[j].status = 'skipped';
-          }
-          break;
+          // If continueOnError is true, continue to next step
         }
       }
 

--- a/src/lib/collections-service.ts
+++ b/src/lib/collections-service.ts
@@ -26,8 +26,11 @@ export const collectionsService = {
 
     collection.requests.push(requestToAdd);
     collection.updatedAt = new Date();
-    
-    await db.collections.update(collectionId, collection);
+
+    await db.collections.update(collectionId, {
+      requests: collection.requests,
+      updatedAt: collection.updatedAt
+    });
   },
 
   async removeRequestFromCollection(collectionId: string, requestId: string) {
@@ -36,8 +39,11 @@ export const collectionsService = {
 
     collection.requests = collection.requests.filter(r => r.id !== requestId);
     collection.updatedAt = new Date();
-    
-    await db.collections.update(collectionId, collection);
+
+    await db.collections.update(collectionId, {
+      requests: collection.requests,
+      updatedAt: collection.updatedAt
+    });
   },
 
   async updateCollection(collectionId: string, updates: Partial<Collection>) {

--- a/src/lib/http-client.ts
+++ b/src/lib/http-client.ts
@@ -13,6 +13,8 @@ interface HttpClientResponse {
   statusText: string;
   headers: Record<string, string>;
   data: any;
+  time: number; // milliseconds
+  size: number; // bytes
 }
 
 class HttpClient {
@@ -33,26 +35,40 @@ class HttpClient {
       }
     }
 
+    const startTime = Date.now();
+
     try {
       const response = await axios(config);
-      
+      const endTime = Date.now();
+      const time = endTime - startTime;
+      const size = JSON.stringify(response.data).length;
+
       return {
         status: response.status,
         statusText: response.statusText,
         headers: response.headers as Record<string, string>,
         data: response.data,
+        time,
+        size,
       };
     } catch (error: any) {
+      const endTime = Date.now();
+      const time = endTime - startTime;
+
       // Handle error responses
       if (error.response) {
+        const size = JSON.stringify(error.response.data).length;
+
         return {
           status: error.response.status,
           statusText: error.response.statusText,
           headers: error.response.headers as Record<string, string>,
           data: error.response.data,
+          time,
+          size,
         };
       }
-      
+
       // Handle network errors
       throw new Error(error.message || 'Network error occurred');
     }

--- a/src/lib/search-service.ts
+++ b/src/lib/search-service.ts
@@ -1,5 +1,5 @@
-import { db, Collection, HistoryItem } from './db';
-import { Request } from '@/types';
+import { db, HistoryItem } from './db';
+import { Request, Collection } from '@/types';
 
 export interface SearchResult {
   id: string;

